daemon/cmd/status.go

```diff
@@ -578,33 +578,25 @@ func (d *Daemon) startStatusCollector(cleaner *daemonCleanup) {
 			Name: "kvstore",
 			Probe: func(ctx context.Context) (interface{}, error) {
 				if option.Config.KVStore == "" {
-					return models.StatusStateDisabled, nil
+					return &models.Status{State: models.StatusStateDisabled}, nil
```

Comment:
This is arguably more correct, given that it explicitly conveys that the subsystem is disabled, rather than claiming it is OK, and reporting `Disabled` in the message. However, if you feel it as a breaking change, then it can be easily modified to preserve the previous behavior.